pallets/transfer-allowlist/src/mock.rs

```diff
@@ -0,0 +1,222 @@
+}
+
+impl frame_system::Config for Runtime {
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type AccountId = MockAccountId;
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockHashCount = BlockHashCount;
+	type BlockLength = ();
+	type BlockNumber = u64;
+	type BlockWeights = ();
+	type DbWeight = ();
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type Header = Header;
+	type Index = u64;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+	type OnKilledAccount = ();
+	type OnNewAccount = ();
+	type OnSetCode = ();
+	type PalletInfo = PalletInfo;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeOrigin = RuntimeOrigin;
+	type SS58Prefix = SS58Prefix;
+	type SystemWeightInfo = ();
+	type Version = ();
+}
+
+type Balance = u64;
+
+// Used to handle reserve/unreserve for allowance creation.
+// Loosely coupled with transfer_allowlist
+impl pallet_balances::Config for Runtime {
+	type AccountStore = System;
+	type Balance = u64;
+	type DustRemoval = ();
+	type ExistentialDeposit = ConstU64<1>;
+	type MaxLocks = ();
+	type MaxReserves = ConstU32<50>;
+	type ReserveIdentifier = [u8; 8];
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
+}
+
+pub type MockAccountId = u64;
+
+parameter_types! {
+	  pub const TreasuryPalletId: PalletId = PalletId(*b"treasury");
+	  pub const Admin: u64 = 1;
+}
+
+impl SortedMembers<u64> for Admin {
+	fn sorted_members() -> Vec<u64> {
+		vec![1]
+	}
+}
+
+// pallet fees takes a treasury impl as assoc type
+impl pallet_treasury::Config for Runtime {
+	type ApproveOrigin = EnsureSignedBy<Admin, u64>;
+	type Burn = ();
+	type BurnDestination = ();
+	type Currency = Balances;
+	type MaxApprovals = ();
+	type OnSlash = Treasury;
+	type PalletId = TreasuryPalletId;
+	type ProposalBond = ();
+	type ProposalBondMaximum = ();
+	type ProposalBondMinimum = ();
+	type RejectOrigin = EnsureSignedBy<Admin, u64>;
+	type RuntimeEvent = RuntimeEvent;
+	type SpendFunds = ();
+	type SpendOrigin = EnsureSignedBy<Admin, u64>;
+	type SpendPeriod = ();
+	type WeightInfo = ();
+}
+
+parameter_types! {
+	  pub const DefaultFeeValue: Balance = 1;
+}
+
+// pallet fees depends on authorship being configured for runtime.
+// Tight coupling--no assoc type for fees
+impl pallet_authorship::Config for Runtime {
+	type EventHandler = ();
+	type FilterUncle = ();
+	type FindAuthor = ();
+	type UncleGenerations = ();
+}
+
+// used to set/retrieve reserve fee amount
+// so we can surface this to the frontend
+// actual reserve/unreserve handled by reserve currency type
+impl pallet_fees::Config for Runtime {
+	type Currency = Balances;
+	type DefaultFeeValue = DefaultFeeValue;
+	type FeeChangeOrigin = EitherOfDiverse<EnsureRoot<Self::AccountId>, EnsureSignedBy<Admin, u64>>;
+	type FeeKey = FeeKey;
+	type RuntimeEvent = RuntimeEvent;
+	type Treasury = Treasury;
+	type WeightInfo = ();
+}
```

Comment:
Yeah having the mocks would definitely be cleaner, I think it would be best to switch to the mocks after benchmarking is done so that the weight/time from those dependencies are taken into account (though in this case it shouldn't really affect the benchmarks much if at all), then switch to use the mocks so the tests aren't coupled with the other pallet dependencies/changes.

Reply:
>I think it would be best to switch to the mocks after benchmarking is done so that the weight/time from those dependencies are taken into account

Mocks will run only under `test`. The real benchmarks that generates the weights will run without `test` and will use the correct implementations coming from the runtime. So there is no issue with benchmark times using trait mocks.

Anyway, no hurry!